### PR TITLE
[SYCL] Add XFAIL trackers for several E2E tests marked with XFAIL

### PR DESCRIPTION
**Pull request by lbushi25**
This PR is the first in a series of PR's to add XFAIL tracker issues for several tests marked as XFAIL on certain platforms.
For XFAIL's on a hip_nvidia platform, we simply remove the XFAIL entirely as we do not test the configuration anymore.
Also, this PR deleted 6 tests failing everywhere because of an issue that will not be fixed.

### CODE REVIEW

File: sycl/test-e2e/Basic/span.cpp [hunk 1]
```diff
@@ -3,6 +3,7 @@
 //
 // Fails to release USM pointer on HIP for NVIDIA
 // XFAIL: hip_nvidia
+// XFAIL-TRACKER: https://github.com/intel/llvm/issues/14404
```
Comment (aelovikov-intel) [inline sycl/test-e2e/Basic/span.cpp:6]: Is this correct? The issue is about AMD HIP, not NVidia.

File: sycl/test-e2e/InvokeSimd/Spec/ImplicitSubgroup/tuple.cpp [hunk 2]
```diff
@@ -1,5 +1,6 @@
 // TODO: enable when Jira ticket resolved
 // XFAIL: *
+// XFAIL-TRACKER: https://jira.devtools.intel.com/browse/GSD-4509
```
Comment (sarnex) [inline sycl/test-e2e/InvokeSimd/Spec/ImplicitSubgroup/tuple.cpp:3]: GSD-4509 is never going to be fixed, so these tests are never goign to work. can we just delete all tests failing based of `GSD-4509`? thanks

**Review by aelovikov-intel — COMMENTED** [review]
If nobody is working on hip_nvidia and that isn't tested anywhere, maybe we should remove `XFAIL`s instead?

File: sycl/test-e2e/Basic/span.cpp (hunk 1 above)
Comment (lbushi25) [reply sycl/test-e2e/Basic/span.cpp]: Thanks for the catch, fixed it!

File: sycl/test-e2e/InvokeSimd/Spec/ImplicitSubgroup/tuple.cpp (hunk 2 above)
Comment (lbushi25) [reply sycl/test-e2e/InvokeSimd/Spec/ImplicitSubgroup/tuple.cpp]: Done.

File: sycl/test-e2e/Basic/span.cpp (hunk 1 above)
Comment (aelovikov-intel) [reply sycl/test-e2e/Basic/span.cpp]: I don't see how this issue is useful though. If the HW is really unsupported, it should be a `REQUIRES: !arch-<something>` or something like that.

Or, going deeper, I don't see a value in just adding a bunch of formal links if they don't become actionable.

\+ @AlexeySachkov

**Review by sarnex — APPROVED** [review]
esimd lgtm

File: sycl/test-e2e/Basic/span.cpp (hunk 1 above)
Comment (AlexeySachkov) [reply sycl/test-e2e/Basic/span.cpp]: I tend to agree with Andrei here. If we don't support this configuration (and AFAIK, we don't), then there is no reason for us to maintain those `XFAIL`s, we can just drop them.

File: sycl/test-e2e/Basic/queue/release.cpp [hunk 3]
```diff
@@ -2,6 +2,7 @@
 // RUN: env SYCL_UR_TRACE=2 %{run} %t.out | FileCheck %s %if !windows %{--check-prefixes=CHECK-RELEASE%}
 //
 // XFAIL: hip_nvidia
+// XFAIL-TRACKER: https://github.com/intel/llvm/issues/16197
```
Comment (aelovikov-intel) [inline sycl/test-e2e/Basic/queue/release.cpp:5]: Shouldn't we remove `XFAIL` instead?

Comment (lbushi25) [reply sycl/test-e2e/Basic/queue/release.cpp]: > Shouldn't we remove `XFAIL` instead?

Yes, this one slipped through the cracks. Fixed it now!

**Review by aelovikov-intel — APPROVED** [review]

**Pull request merged**